dask_expr/_collection.py

```diff
@@ -1228,6 +1228,8 @@ def set_index(
         upsample: float = 1.0,
         partition_size: float = 128e6,
     ):
+        if isinstance(other, list) and len(other) == 1:
+            other = other[0]
```

Comment:
This will miss the `FrameBase` check below, is that okay? See dask/dask checks for Sequence, so guessing its some similar error checking missing?
https://github.com/dask/dask/blob/f2d448e7a8561c2488806cb398470459984f11b0/dask/dataframe/core.py#L5406

Reply:
Yep that's fine, length one lists can be supported